esp/src/src/WsTopology.ts

```diff
@@ -39,31 +40,30 @@ var TpLogFileStore = declare([Memory, Evented], {
             }).then(lang.hitch(this, function (response) {
                 var data = [];
                 if (lang.exists("TpLogFileResponse.LogData", response)) {
+                    var columns = response.TpLogFileResponse.LogFieldNames.Item;
                     this.lastPage = response.TpLogFileResponse.LogData;
                     this.emit("pageLoaded", this.lastPage);
                     arrayUtil.forEach(response.TpLogFileResponse.LogData.split("\n"), function (item, idx) {
                         if (options.start === 0 || idx > 0) {
                             //  Throw away first line as it will probably only be a partial line  ---
                             var itemParts = item.split(" ");
-                            var lineNo, audience, time, pid, tid, date, details;
-                            if (itemParts.length) lineNo = nextItem(itemParts);
-                            if (itemParts.length) audience = nextItem(itemParts);
-                            if (itemParts.length) date = nextItem(itemParts);
-                            if (itemParts.length) time = nextItem(itemParts);
-                            if (itemParts.length) pid = nextItem(itemParts);
-                            if (itemParts.length) tid = nextItem(itemParts);
-                            if (itemParts.length) details = itemParts.join(" ");
+                            var tempObj = {
+                                __hpcc_id: response.TpLogFileResponse.PageNumber + "_" + idx
+                            };
 
-                            data.push({
-                                __hpcc_id: response.TpLogFileResponse.PageNumber + "_" + idx,
-                                lineNo: lineNo,
-                                audience: audience,
-                                date: date,
-                                time: time,
-                                pid: pid,
-                                tid: tid,
-                                details: details
-                            });
+                            for  (var i = 0; i < columns.length; ++i) {
+                                var cleanName = Utility.removeSpecialCharacters(columns[i]);
+                                var value;
```

Comment:
Assign a default value of `""`?

Reply:
Also you can make `const` rather than var.

Reply:
Cannot make it a `const` it will not allow a reassign